core/internal/api/send.go

```diff
@@ -19,8 +19,10 @@ func (client *Client) Send(req *Request) (*http.Response, error) {
 	}
 
 	for headerKey, headerValue := range req.Headers {
-		retryableReq.Header.Add(headerKey, headerValue)
+		retryableReq.Header.Set(headerKey, headerValue)
 	}
```

Comment:
nit: what do you think of having this one as an helper function, just for consistency?